R/data.table.R

```diff
@@ -1739,22 +1739,48 @@ replace_dot_alias = function(e) {
         GForce = FALSE
       } else {
         # Apply GForce
+        noCall_noVars = function(expr) (!is.call(expr) || length(all.vars(expr, max.names=1L, unique=FALSE))==0L) && !dotN(expr)
+        .gshift_ok = function(q) {
+          q = match.call(shift, q)
+          noCall_noVars(q[["n"]]) &&
+            noCall_noVars(q[["fill"]]) &&
+            noCall_noVars(q[["type"]]) &&
+            is.null(q[["give.names"]])
+        }
+        .ghead_ok = function(q) {
+          length(q) == 3L &&
+            length(q[[3L]]) == 1L &&
+            noCall_noVars(q[[3L]])
+        }
+        ".g[_ok" = function(q, x) {
+          length(q) == 3L &&
+            length(q[[3L]]) == 1L &&
+            noCall_noVars(q[[3L]]) &&
+            (q[[1L]] == "[" || (q[[1L]] == "[[" && eval(call('is.atomic', q[[2L]]), envir=x))) &&
+            eval(q[[3L]], parent.frame(3L)) > 0L
+        }
+        .gweighted.mean_ok = function(q, x) { #3977
+          q = match.call(function(x, w, ..., na.rm=FALSE) {}, q)
```

Comment:
this is the signature of `stats:::weighted.mean.default()`, right? And we are just avoiding that to avoid using `:::` in the package?

Assuming that's right I'd do one of:

 1. Write a comment making that explicit
 2. use `get("weighted.mean.default", asNamespace("stats"))` to avoid the `R CMD check` note about `:::`

Reply:
Oh, I see `...` was already added -- why not `match.call(gweighted.mean, q)`?

Reply:
Yes. Good call.